common/scala/src/main/scala/whisk/common/Logging.scala

```diff
@@ -156,6 +168,18 @@ private object Emitter {
 
 case class LogMarkerToken(component: String, action: String, state: String) {
     override def toString() = component + "_" + action + "_" + state
+
+    def asFinish = copy(state = LoggingMarkers.finish)
+    def asError = copy(state = LoggingMarkers.error)
+}
+
+object LogMarkerToken {
+    def parse(s: String) = {
+        // Per convention the components are guaranteed to not contain '_'
```

Comment:
There isn't one deliberately. A runtime check will likely introduce quite some overhead for the markers we generate on the fly.